### PR TITLE
Prevent suffocation while using `/rtp`

### PR DESCRIPTION
**Pull request by USA-RedDragon**
Added some code to check the two blocks above the teleportation block for air to avoid situations where a player might suffocate on `/rtp` use.

### CODE REVIEW

File: common/src/main/java/net/william278/huskhomes/util/UnsafeBlocks.java [hunk 1]
```diff
@@ -49,6 +52,19 @@ public boolean isUnsafe(@NotNull String blockId) {
...
+    /**
+     * Returns if the block, by provided identifier, is unsafe to stand in
+     *
+     * @param blockId The block identifier (e.g. {@code minecraft:stone})
+     * @return {@code true} if the block is on the unsafe blocks list, {@code false} otherwise
+     */
+    public boolean isUnsafeToOccupy(@NotNull String blockId) {
+        if (!blockId.startsWith(Key.MINECRAFT_NAMESPACE + ":")) {
+            blockId = Key.MINECRAFT_NAMESPACE + ":" + blockId;
+        }
```
Comment (WiIIiam278) [inline common/src/main/java/net/william278/huskhomes/util/UnsafeBlocks.java:64]: Could we deduplicate this stuff into a private `formatBlockId` method here?

**Review by WiIIiam278 — CHANGES_REQUESTED** [review]
Great work :)

It feels _slightly_ wasteful to have to instantiate multiple cursors -- but it's reasonable in this instance. Just a few small nitpicks, really.

**Comment by USA-RedDragon** [comment]
> It feels slightly wasteful to have to instantiate multiple cursors

Yeah, I'm not familiar with the Sponge API at all, just kinda copied the existing lines and modified them 😁 

Thanks for the review. I'll make those changes tomorrow and get them pushed.

File: bukkit/src/main/java/net/william278/huskhomes/util/BukkitSafetyResolver.java [hunk 2]
```diff
@@ -72,12 +72,16 @@ private Optional<Location> findSafeLocationNear(@NotNull Location location, @Not
                 if (x < 0 || x >= 16 || z < 0 || z >= 16) {
                     continue;
                 }
-                final int y = Math.max((minY + 1), snapshot.getHighestBlockYAt(x, z));
-                final Material blockType = snapshot.getBlockType(chunkX, y, chunkZ);
-                if (isBlockSafe(blockType.getKey().toString())) {
+                final int y = Math.max((minY + 1), snapshot.getHighestBlockYAt(x, z)) + 1;
+                final Material blockType = snapshot.getBlockType(x, y-1, z);
```
[the remark this answers is not in the document]
Comment (USA-RedDragon) [reply bukkit/src/main/java/net/william278/huskhomes/util/BukkitSafetyResolver.java]: fixed in rebase of https://github.com/WiIIiam278/HuskHomes/pull/445/commits/eda37687d0442c1aebb67ce8d13ed3414095b76e

File: common/src/main/java/net/william278/huskhomes/util/UnsafeBlocks.java (hunk 1 above)
Comment (USA-RedDragon) [reply common/src/main/java/net/william278/huskhomes/util/UnsafeBlocks.java]: https://github.com/WiIIiam278/HuskHomes/pull/445/commits/1941b3785ed384bac38c1b426ac265641ff0fb5e

**Review by WiIIiam278 — APPROVED** [review]
Nice work :)

**Pull request merged**